src/pages/manage_dac/ManageDac.jsx

```diff
@@ -70,12 +74,47 @@ export const ManageDac = function ManageDac() {
     ]);
   }, [reloadDacList, reloadUserRole]);
 
+  useEffect(() => {
+    const init = async () => {
+      try {
+        setIsLoading(true);
+        const daaList = await DAA.getDaas();
+        setDaas(daaList);
+        setIsLoading(false);
+      } catch (error) {
+        Notifications.showError({
+          text: 'Error: Unable to retrieve current DAAs from server',
+        });
+        setIsLoading(false);
+      }
+    };
+    init();
+  }, []);
+
   const handleDeleteDac = async () => {
-    let status;
+    let statusDac;
+    const matchingDaas = daas.filter(daa => daa.initialDacId === selectedDac.dacId && daa.daaId !== 17);
+    const deletePromises = matchingDaas.map(daa => DAA.deleteDaa(daa.daaId));
+    const deleteResponses = await Promise.all(deletePromises);
+    const failedDeletes = deleteResponses.filter(resp => resp.status !== 200);
+    const fullSelectedDac = await DAC.get(selectedDac.dacId); // THIS IS HACKY! BUT ALL I CAN DO
+    if (fullSelectedDac.associatedDaa.daaId === 17) {
+      await DAA.deleteDacDaaRelationship(17, selectedDac.dacId).then((resp) => {
+        if (resp.status !== 200) {
+          failedDeletes.push(resp);
+        }
+      });
+    }
```

Comment:
Additionally, I think we can move the bulk of the logic to the back end - it doesn't make sense to me to try and manage this at the UI.

Reply:
Sure! I had just ended up handling it here because I was unable to delete my testing DACs that I was creating to test this functionality. Here's the ticket: https://broadworkbench.atlassian.net/browse/DCJ-469